### ISSUE TITLE
Make outline work

### ISSUE THREAD

**Issue by svenefftinge** [yang-tools/yang-eclipse#3]
The hierachy in the outline in Eclipse LSP4E is based on nested ranges. 
Since in yang.lsp we return the significiant range (the range of the identifier), they are not nested. 
We should check if reveal within monaco (Theia & VsCode) works well with full ranges, too.
See also https://bugs.eclipse.org/bugs/show_bug.cgi?id=520770

**Comment by JanKoehnlein** [comment]
I've tried with VSCode:
I had to install the (Code Outline)[https://marketplace.visualstudio.com/items?itemName=patrys.vscode-code-outline] extension to see something a code outline. 
Extending the regions works in general, but of course the entire region of the element is selected when the node is clicked. Given that the children are also sorted in alphabetic order, the selection feels pretty quirky when we step through the children. Worst of all, the caret is placed at the end of the region. The code lens interrupts the selection visually in an ugly way. 

Most of these problems seem to be caused by the Code Outline extension. I have installed TypeScript Hero and that one behaves in the same bad way: Full region select, caret placed at the end of the region. Pretty unusable. But it TS Hero contributes its own additional code outline that does not select, places the cursor at the beginning at the region and does nothing when selecting child nodes (in fact it shows less children than the other outline, e.g. nothing within functions).

Given the other outline issues, I am inclined to implement our own outline.
